Port Current 1.19.3 Release to 1.20.1

BlueAgent opened the pull request:
This is a port of the current release of Haema to 1.20.1 because a friend really wanted to play with Haema on 1.20.1.

Some notes for the port (some of which I think you have already noticed in the rewrite):

- Added included dep MixinExtras to be able to use the `WrapWithCondition` annotation to avoid having to use a `Redirect`.
  - The minecraft rendering code changed to no longer manually setting the texture, instead passing in the texture (which does end up setting the texture internally, but it means we don't have to set the texture back again).
- Disabled Rats Compat. (It's not available for 1.20.1 yet.)
- `DamageSources` changed quite a bit and now have `DamageType` so added a `DamageTypeModule`.
  - Some of damage source checks were changed to directly check the damage type. Ideally we should use the new damage type tags, possibly adding our own ones.

I know you're currently working on a rewrite and the contributing guidelines say to wait for you to update the guidelines before doing code PRs so it's okay if you don't want to release a official 1.20.1 version of the old code. Feel free to close this if so.

BlueAgent committed:
chore: update gradle wrapper to v8.5

BlueAgent committed:
chore: update dependencies to their mc 1.20.1 versions

BlueAgent committed:
chore(wip): update to 1.20.1

BlueAgent committed:
chore: move patchouli book to assets as required

BlueAgent committed:
fix: full and half blood icons swapped

BlueAgent committed:
fix: weird classloading issue when setting accessible

BlueAgent committed:
fix: origins vampire power failing to read when entity is null

BlueAgent committed:
chore: bump version

BlueAgent committed:
chore: declare dependency on mixin extras

BlueAgent committed:
chore: move mixin extras version to gradle properties

BlueAgent commented:
Hey there's some bugs in the port (that my friend @snowwymira found) so gotta fix them first aaaaah. It's a bit challenging to debug since pehkui doesn't seem to be able to run in-dev sadly (due to using `remap=false` for mixins for the cross-version compatibility on quilt/fabric).

- [x] Ritual Table (Ability) Screen is empty.
- [ ] Mist mode does not hide player (amusingly they are squished flat xD).
  - Turns out this might be due to some cross mod interaction in a modpack, it doesn't repro with just haema on it's own or when adding sodium + iris + indium either.

williambl commented:
oh this is awesome thank you. especially since i have so little time that the rewrite is gonna take like a hundred years lol.

i believe if you switch to latest fabric loader you won't need to include mixin extras manually by the way since they added it into floader.

feel free to ping me on discord when this is ready for review :)

BlueAgent committed:
fix: ritual screen rendering

BlueAgent committed:
chore: bump version

BlueAgent committed:
chore: don't include mixin extras because is already included by fabric loader

BlueAgent commented:
Oh that's nice that Mixin Extras is included in the loader, I didn't know! Made it no longer included just a normal dependency now.

- Fixed the Ritual Table rendering being broken in the initial port attempt. Also made the clickable area for the abilities more closely fit the rendering too.
- The mist bug with the player still rendering in-world I couldn't reproduce outside of the modpack, even with sodium, iris and indium installed. So we think it's some cross mod interaction possibly.

This should be good for a review now I'll ping you on discord as well.

BlueAgent committed:
fix: haema damage types not bypassing armour

BlueAgent committed:
fix: warning about no InSize uniform

Caused by said uniform being unused and thus optimized out. The fragment shaders were not using oneTexel from the sobel vertex shader so switched to using blit vertex shader and removed the InSize uniform.

BlueAgent committed:
chore: bump version
